### ISSUE TITLE
Sending logs to Loki - how to basic auth

### ISSUE THREAD

**Issue by vlcinsky** [fluent/fluent-bit#2735]
## Bug Report

**Describe the bug**
We did not find a way to specify username and password for Loki endpoint requiring basic auth.

Using the form `https://username:password@host/url` does not work.

**Expected behavior**
Either support the `https://username:password@host/url` url pattern or provide configuration parameters for basic auth.

**Comment by Goshander** [comment]
Same problem, it is last step before to perfect use Loki in production. And also support tls is not be superflous)

**Comment by MezNyx** [comment]
I am also having this issue when trying to use the plugin to send to grafana cloud via their endpoint of: 
```
[OUTPUT]
  name loki
  match *
  host <username>:<apikey>@logs-prod-us-central1.grafana.net/loki/api/v1/push
  port 443
```

I get the following:
[2020/11/18 17:14:35] [ warn] net_tcp_fd_connect: getaddrinfo(host='<username>:<apikey>@logs-prod-us-central1.grafana.net/loki/api/v1/push'): Name or service not known
[2020/11/18 17:14:35] [error] [io] connection #34 failed to: <username>:<apikey>@logs-prod-us-central1.grafana.net/loki/api/v1/push:443
[2020/11/18 17:14:35] [error] [output:loki:loki.0] no upstream connections available

**Comment by nweseloh** [comment]
Same issue for us, it is preventing us from using the great loki output support plugin in production with grafana cloud. 

As far as I see, the underlying flb_http_client already supports basic auth + HTTPs scheme, so it might be a fairly easy fix.

**Comment by gerrit-amagno** [comment]
Fluent Bit works like a charm but I need support for basic auth and https to use grafana cloud, too. A fix would be perfect!

**Comment by edsiper** [comment]
Thanks everyone for raising this issue.

I've pushed fix d59f1bad12e52eb9a8a02f386a7f2beb5a7aa7c0 to add HTTP Basic Auth. The new options are called ```http_user``` and ```http_passwd```. 

The enhancement will be part of the next release today.
